Barotrauma/BarotraumaShared/SharedSource/Items/Components/Machines/Fabricator.cs

```diff
@@ -365,24 +372,35 @@ static void onItemSpawned(Item spawnedItem, Character user)
 
         partial void UpdateRequiredTimeProjSpecific();
 
-        private bool CanBeFabricated(FabricationRecipe fabricableItem)
+        private bool CanBeFabricated(FabricationRecipe fabricableItem, Dictionary<String, List<Item>> availableIngredients)
         {
             if (fabricableItem == null) { return false; }
-            List<Item> availableIngredients = GetAvailableIngredients();
-            return CanBeFabricated(fabricableItem, availableIngredients);
-        }
 
-        private bool CanBeFabricated(FabricationRecipe fabricableItem, IEnumerable<Item> availableIngredients)
-        {
-            if (fabricableItem == null) { return false; }            
-            foreach (FabricationRecipe.RequiredItem requiredItem in fabricableItem.RequiredItems)
+            return fabricableItem.RequiredItems.All(requiredItem =>
             {
-                if (availableIngredients.Count(it => IsItemValidIngredient(it, requiredItem)) < requiredItem.Amount)
+                int availablePrefabsAmount = 0;
+                foreach (ItemPrefab requiredPrefab in requiredItem.ItemPrefabs)
                 {
-                    return false;
+                    if (!availableIngredients.ContainsKey(requiredPrefab.Name)) { continue; }
+
+                    var availablePrefabs = availableIngredients[requiredPrefab.Name];
+                    foreach (Item availablePrefab in availablePrefabs)
+                    {
+                        if (availablePrefab.Condition / availablePrefab.Prefab.Health >= requiredItem.MinCondition &&
+                            availablePrefab.Condition / availablePrefab.Prefab.Health <= requiredItem.MaxCondition)
+                        {
+                            availablePrefabsAmount++;
+                        }
+
+                        if (availablePrefabsAmount > requiredItem.Amount)
```

Comment:
If I'm not mistaken, this should be `availablePrefabsAmount >= requiredItem.Amount`. Fixed on our private repo.